mne/viz/epochs.py

```diff
@@ -169,14 +168,14 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
     if combine is not None:
         ts_args["show_sensors"] = False
 
-    if picks is None or not _is_numeric(picks):
-        picks = _picks_to_idx(epochs.info, picks)
-        if group_by is None:
-            logger.info("No picks and no groupby, showing the first five "
-                        "channels ...")
-            picks = picks[:5]  # take 5 picks to prevent spawning many figs
-    else:
-        picks = _picks_to_idx(epochs.info, picks)
+    picks = [picks] if isinstance(picks, str) else picks
```

Comment:
Breaks some ICA stuff

https://travis-ci.org/mne-tools/mne-python/jobs/530103782

You might be able to just do `picks = _picks_to_idx(epochs.info, picks)` directly, it should handle single strings just fine.

Reply:
it's actually the following line that breaks it (the generator expression inside the `all()`).  I'll try to find a better way.

Reply:
My point is really that instead of making this conditional any smarter (which your next commit did I think), you should just do `picks = _picks_to_idx` here, as it should already have all of the smartness built in. Then on the next line you know `picks` is a ndarray of int, and can just check the length.

Reply:
that would change the behavior.  Currently we only truncate to 5 picks if the user passes a channel type.  Under your suggestion, if a user asked for 8 specific channels (by name or by index) we would only return them 5 channels.

Reply:
> Currently we only truncate to 5 picks if the user passes a channel type

I would make it so that `_picks_to_idx` can optionally return how it made the picks (from channel names, channel types, from None, etc.). Then call it with `return_how=True` and check that here.

Reply:
in other words, once you call `_picks_to_idx` you lose information about what they provided to the `picks` argument (specific channel names vs. indices vs. channel types).  I'm not opposed to changing the behavior here --- the way it's set up now seems paternalistic ("surely that's not what you want, here's 5 arbitrary channels instead").  In my opinion, if a user passes `picks='eeg'` they should get GFP by default (just like they get GFP for each channel type when `picks=None` --- which would be consistent with what was done in #6166).  WDYT?

Reply:
oops, missed your response...  so we have 1 proposal for having `_picks_to_idx` keep track of its input type and optionally returning it, and 1 proposal for changing behavior of `plot_image` to return GFP when picks is a channel type.  Votes?

Reply:
@agramfort @jona-sassenhagen ?

Reply:
See reply above, too -- but I'm fine either way. @jona-sassenhagen ?

Either way let's not add any special `picks` processing here (in fact let's remove the special list-making code already here), and instead have `_picks_to_idx` return more information if it's needed, rather than adding redundant code here for processing different variants of `picks`.

Reply:
>In my opinion, if a user passes picks='eeg' they should get GFP by default (just like they get GFP for each channel type when picks=None --- which would be consistent with what was done in #6166). WDYT?

Or the mean?..

In principle, this might be the better choice. Note I set it to 5 because originally, it would plot ALL channels if you forgot picks, which would open a lot of axes. The 5 chans was so you wouldn't have 300 axes open after doing `epochs..plot_image()`.

Reply:
OK.  I prefer `picks='eeg'` (or any channel type) returning the GFP since it's consistent with what @jona-sassenhagen implemented for evokeds in #6211 (see [here](https://github.com/mne-tools/mne-python/pull/6211/files#diff-89af0f6346b5851261413eb035d5a650R2140)). And since @larsoner says he's fine either way, I'll go ahead and implement that (shouldn't be too much work) and let @agramfort weigh in afterwards.

This will be an API change (same args, different behavior)... my instinct is that it's edge-case behavior anyway (not *exactly* a bug, but getting plots of 5 arbitrary channels was probably not desired behavior for users anyway) so maybe OK to skip the warning cycle?  WDYT?

Reply:
I think GFP is better for lines than for images because the GFP, not starting at 0, is squished to a small subset of the color range for images. You simply see less. This is not so much an issue for line plots.

Reply:
ah, good point.

Reply:
> I think GFP is better for lines than for images because the GFP, not starting at 0, is squished to a small subset of the color range for images.

When doing GFP (and really any one-sided data), we should probably switch the colormap to `viridis` or some other sequential rather than diverging colormap. It will help ameliorate this problem at least somewhat.

Knowing the complexity of our `viz` code it's entirely possible this will not be an easy change, though, so no need to do it here unless it turns out to be easy and does work well.

Reply:
For all-positive images like GFP, we actually use `Reds` here. But maybe Viridis would be better still.